docs/aws/audit/redshiftmonitoring/rules/redshift_nodes_counts.mdx

```diff
@@ -24,6 +24,270 @@ CBP
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. Install and configure AWS CLI: Before you can start using AWS CLI, you need to install it on your local machine. You can download it from the official AWS website. After installation, you need to configure it with your AWS account credentials. You can do this by running the command `aws configure` and then entering your AWS Access Key ID, Secret Access Key, Default region name, and Default output format when prompted.
+
+2. List all Redshift clusters: Use the following command to list all the Redshift clusters in your AWS account.
+
+   ```
+   aws redshift describe-clusters
+   ```
+
+   This command will return a JSON output with details of all the Redshift clusters.
+
+3. Check the number of nodes in each cluster: In the JSON output from the previous step, look for the 'NumberOfNodes' field. This field indicates the number of nodes in the cluster. 
+
+   ```
+   aws redshift describe-clusters --query 'Clusters[*].[ClusterIdentifier,NumberOfNodes]'
+   ```
+
+   This command will return a list of all clusters along with their node count.
+
+4. Compare the number of nodes with the limit: AWS has a default limit of 50 nodes per cluster for Redshift. If the 'NumberOfNodes' field in the output is close to or at this limit, then you are at the Redshift nodes limit.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3): To interact with AWS services, you need to install and configure Boto3. You can install it using pip:
+
+   ```
+   pip install boto3
+   ```
+   Then, configure your AWS credentials to enable Boto3 to communicate with AWS services. You can do this by setting the following environment variables:
+
+   ```
+   AWS_ACCESS_KEY_ID = 'your_access_key'
+   AWS_SECRET_ACCESS_KEY = 'your_secret_key'
+   AWS_SESSION_TOKEN = 'your_session_token' # optional
+   ```
+
+2. Use Boto3 to interact with AWS Redshift: Once Boto3 is installed and configured, you can use it to interact with AWS Redshift. Here is a sample Python script that lists all Redshift clusters and their nodes:
+
+   ```python
+   import boto3
+
+   def list_redshift_clusters():
+       redshift = boto3.client('redshift')
+       response = redshift.describe_clusters()
+       for cluster in response['Clusters']:
+           print(f"Cluster: {cluster['ClusterIdentifier']}, Nodes: {cluster['NumberOfNodes']}")
+
+   list_redshift_clusters()
+   ```
+   This script uses the `describe_clusters` method of the Redshift client to retrieve information about all Redshift clusters. It then prints the identifier and the number of nodes for each cluster.
+
+3. Check the Redshift Nodes Limit: AWS has a default limit on the number of nodes that you can have in a Redshift cluster. You can check this limit by calling the `describe_account_attributes` method of the Redshift client:
+
+   ```python
+   def check_redshift_nodes_limit():
+       redshift = boto3.client('redshift')
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>